model/pipeline/src/main/proto/metrics.proto

```diff
@@ -52,38 +55,157 @@ message Annotation {
   string value = 2;
 }
 
-// Populated MonitoringInfoSpecs for specific URNs.
-// Indicating the required fields to be set.
-// SDKs and RunnerHarnesses can load these instances into memory and write a
-// validator or code generator to assist with populating and validating
-// MonitoringInfo protos.
+// A set of well known MonitoringInfo specifications.
 message MonitoringInfoSpecs {
   enum Enum {
-    // TODO(BEAM-6926): Add the PTRANSFORM name as a required label after
-    // upgrading the python SDK.
-    USER_COUNTER = 0 [(monitoring_info_spec) = {
-      urn: "beam:metric:user",
-      type_urn: "beam:metrics:sum_int_64",
+    // Represents an integer counter where values are summed across bundles.
+    USER_SUM_INT64 = 0 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:v1",
+      type: "beam:metrics:sum_int64:v1",
       required_labels: ["PTRANSFORM", "NAMESPACE", "NAME"],
       annotations: [{
         key: "description",
-        value: "URN utilized to report user numeric counters."
+        value: "URN utilized to report user metric."
       }]
     }];
 
-    ELEMENT_COUNT = 1 [(monitoring_info_spec) = {
+    // Represents a double counter where values are summed across bundles.
+    USER_SUM_DOUBLE = 1 [(monitoring_info_spec) = {
+      urn: "beam:metric:user:v1",
```

Comment:
The URN represents the unique identifier in the "semantic meaning" space of the counter. So saying that this is a user counter and that there are many types makes sense to me.

Reply:
I suppose I still haven't bought into the idea that the whole set of "user counters" belongs to a single semantic identifier on par with each individual system counter like msecs or element counts--all we really need is a way to not have names between the two collide--but if we're going to go this route I don't think there's value in enumerating them here (vs. simply saying "user counter" can use any type).

Reply:
For the Go short id generation, I'm relying on the label set to be unique, and map to the semantic URN and Type. That is, I'm assuming for a counter with a given namespace, name, and ptransform, will always be the same URN and Type, and should use the same short id.

Reply:
I enumerate them so that its easier to write validation logic in the future that uses the monitoring specs.

Also I don't believe user counters can have any type as the progress type doesn't make much sense.

Reply:
Made all the URNs unique and added a test to make sure that they remain unique.